### ISSUE TITLE
BUG : Error due to RSCrashReporterEvent in CrashReporter

### ISSUE THREAD

**Issue by Lunacat78** [rudderlabs/rudder-sdk-ios#432]
When running my unit test builds on a remote runner, I'm encountering the following trace:

```
/Users/builder/Library/Developer/Xcode/DerivedData/project-cwqhmjydvvzabdarkejcyutjmhcf/SourcePackages/checkouts/metrics-reporter-ios/Sources/Classes/Plugins/CrashReporter.swift:36:31: cannot find type 'RSCrashReporterEvent' in scope

    func notifyCrash(_ event: RSCrashReporterEvent?, withRequestPayload requestPayload: [AnyHashable: Any]?) {
                              ^~~~~~~~~~~~~~~~~~~~

/Users/builder/Library/Developer/Xcode/DerivedData/project-cwqhmjydvvzabdarkejcyutjmhcf/SourcePackages/checkouts/metrics-reporter-ios/Sources/Classes/Plugins/CrashReporter.swift:44:36: cannot find type 'RSCrashReporterEvent' in scope

    func checkIfRudderCrash(event: RSCrashReporterEvent?) -> Bool {
                                   ^~~~~~~~~~~~~~~~~~~~

```
I've tried to manually add RSCrashReporter to my project but that did not change the output.  This began after updating to use SPM and v2.0 of the Rudder SDK.  I'm on the latest version of RS and can successfully use the app (and see output in RS) using the simulator.  The runner is cleared upon each run so clearing the DerivedData folder would not change the outcome.

**Version of the _iOS_ SDK**
iOS SDK v1.24.1

**Comment by pallabmaiti** [comment]
iOS v2 is in beta and is not compatible with RSCrashReporter. 
You can try it out, but we recommend not to use it for production.

**Comment by AlfredReisner** [comment]
Probably this issue is not about v2.0 of the Rudder SDK, but a different version for the MetricsReporter in the SPM package configuration. If I see correctly the Package.swift still uses version 1.1.0 of the MetricsReporter. (not 1.2.0 as for Carthage and CocoaPods).

**Comment by pallabmaiti** [comment]
Can you elaborate on the issue correctly?
Btw, we will update the version of RSCrashReporter in SPM.

**Comment by AlfredReisner** [comment]
I guess updating RSCrashReporter in SPM might solve this issue. (I can´t oversee it correctly, but it seems RSCrashReporterEvent got added, removed or renamed in an updated RSCrashReporter version?)

We can reproduce it on two different machines right now.

Neither SDK v1.23.0 nor v1.24.1 work for us anymore.
SPM checks out RSCrashReporter 1.1.0, MetricsReporter 1.2.0 on both scenarios.

**Comment by pallabmaiti** [comment]
Got it. We will release a fix soon.

**Comment by SwiftedMind** [comment]
@pallabmaiti @desusai7 I'm not entirely sure, but I think the problem is something else. SPM already resolves MetricsReporter to version 1.2.0, in which the error occurs, so increasing the version shouldn't change anything.

`MetricsReporter` specifies this dependency:

```swift
.package(name: "RSCrashReporter", url: "https://github.com/rudderlabs/crash-reporter-ios", from: "1.0.0"),
```

Xcode resolves this by adding `RSCrashReporter` version `1.1.0`, which is really strange because there is no SPM-compatible 1.1.0 release tag in that repository. There is only a `v1.1.0` tag, which (as far as I'm aware) should be ignored by SPM, because it has a "v"-prefix.

Additionally, this `1.1.0` version does not seem to have the changes from `1.0.1` applied to it (which fix the compilation error).

So the problem is that Xcode erroneously (I guess?) resolves the wrong SPM version of the `RSCrashReporter` package.

**Comment by desusai7** [comment]
@SwiftedMind @Lunacat78 @AlfredReisner  Thank you for pointing it out. We've deleted the tag `1.1.0` and released version `1.24.2` of the iOS SDK. Please try upgrading to it. 

If you still see that version `1.1.0` of `RSCrashReporter` is being fetched, try clearing the cache in the below folder `~/Library/Caches/org.swift.swiftpm/repositories`

**Comment by SwiftedMind** [comment]
It works now, thanks a lot for the quick fix! 🎉
